hail/src/main/scala/is/hail/expr/ir/EmitCodeBuilder.scala

```diff
@@ -25,14 +25,18 @@ object EmitCodeBuilder {
   }
 }
 
-class EmitCodeBuilder(emb: EmitMethodBuilder[_], var code: Code[Unit]) extends CodeBuilderLike {
+class EmitCodeBuilder(val emb: EmitMethodBuilder[_], var code: Code[Unit]) extends CodeBuilderLike {
   def mb: MethodBuilder[_] = emb.mb
 
   def append(c: Code[Unit]): Unit = {
     code = Code(code, c)
   }
 
-  def result(): Code[Unit] = code
+  def result(): Code[Unit] = {
+    val tmp = code
+    code = Code._empty
+    tmp
+  }
```

Comment:
Not sure this is still necessary. Can revert. Though making sure result is only called once may be a good invariant.

Reply:
Seems like a fine sanity check. The `Code` itself can only be used once, so if you call `result` twice and use it both times, it will catch the error, but this would probably give an easier to understand stack trace.